src/datachain/lib/file.py

```diff
@@ -272,8 +272,12 @@ def read_text(self):
     def save(self, destination: str):
         """Writes it's content to destination"""
         destination = stringify_path(destination)
-        client: Client = self._catalog.get_client(str(destination))
-        client.upload(self.read(), str(destination))
+        client: Client = self._catalog.get_client(destination)
+
+        if client.PREFIX == "file://" and not destination.startswith(client.PREFIX):
+            destination = f"{client.PREFIX}{destination}"
```

Comment:
There is no such thing as relative `file://` URI. The path needs to be converted to an absolute path, and then to an URI.